rustls/src/client/hs.rs

```diff
@@ -142,6 +143,18 @@ pub(super) fn start_handshake(
 
     let random = Random::new(config.provider.secure_random)?;
 
+    let ech_context = match &config.ech_config {
```

Comment:
Style: `as_ref().map()`?

Reply:
I switched to `as_ref`, but using `map` isn't ergonomic here because I want to `?` the `new()` error into an early return and can't do that from within the `map` closure.